Mainnet/AddressMapper/AddressMapper/AddressMapper.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using Stratis.SmartContracts;
+
+public class AddressMapper : SmartContract
+{
+    public Address GetSecondaryAddress(Address primary) => PersistentState.GetAddress($"SecondaryAddress:{primary}");
+    private void SetSecondaryAddress(Address primary, Address secondary) => PersistentState.SetAddress($"SecondaryAddress:{primary}", secondary);
+
+    private MappingInfo GetMappingInfo(Address secondary) => PersistentState.GetStruct<MappingInfo>($"MappingInfo:{secondary}");
+    private void SetMappingInfo(Address secondary, MappingInfo value) => PersistentState.SetStruct($"MappingInfo:{secondary}", value);
+
+    private void ClearMappingInfo(Address secondary) => PersistentState.Clear($"MappingInfo:{secondary}");
+
+    public Address Owner
+    {
+        get => PersistentState.GetAddress(nameof(Owner));
+        private set => PersistentState.SetAddress(nameof(Owner), value);
+    }
+
+    public AddressMapper(ISmartContractState smartContractState, Address owner) : base(smartContractState)
+    {
+        this.Owner = owner;
+    }
+
+    public void MapAddress(Address secondary)
+    {
```

Comment:
Anyone can DoS this contract by attempting to register a secondary address that doesn't belong to them. Although it requires approval, it will prevent the valid owner from being able to make a claim. Theoretically a malicious actor who really wants to prevent someone from registering an address could watch for a rejection tx in a block and repeat the process indefinitely.

Reply:
Identity Server is owner of the contract and Approve/Reject mapping requests based on logged in user decisions. So basically user approve/reject the mapping over Identity Server. 

User wants to associate his desktop address to mobile address so user has to make a call to contract from Desktop wallet by passing mobile wallet address. So for this reason method is not owner only. 

Secondary address is mobile address and it is very hard to know who is the address owner also it is easy to generate new account on mobile app. Theoretically possible but practically very hard abuse the contract. Also DOS attack is going to cost to attacker.

Reply:
I will look is there a better solution.